src/solvers/dgsem_tree/subcell_limiters.jl

```diff
@@ -14,12 +14,14 @@ end
 
 """
     SubcellLimiterIDP(equations::AbstractEquations, basis;
+                      local_minmax_variables_cons = [],
                       positivity_variables_cons = [],
                       positivity_correction_factor = 0.1)
```

Comment:
I just noticed this now:
* Why the `_cons`? Do you plan to support limiting on non-cons variables? If not, I'd remove it as it suggests there might be other suffixes possible
* Why provide the variables as integer indices? This requires an intimate knowledge of memory layout/variable order etc. Why not allow people to say `"p"` or `"rho"`? If you want integer coefficients for efficiency reasons internally, you can always translate variable names into indices in the constructor
* Whatever you do, it seems like the way you specify the variables is not described in the docs. I.e., it is not clear what `local_minmax_variables_cons` and `positivity_variables_cons` expects as inputs - this should be documented in the docstring

Reply:
> Why the _cons? Do you plan to support limiting on non-cons variables? If not, I'd remove it as it suggests there might be other suffixes possible

For the local limiting, my implementation does not support other than conservative variables. That's only the case for global positivity limiting. I added it here as well for consistency. But I can remove it if you want.

> Why provide the variables as integer indices? This requires an intimate knowledge of memory layout/variable order etc. Why not allow people to say "p" or "rho"? If you want integer coefficients for efficiency reasons internally, you can always translate variable names into indices in the constructor

I used the variable names in the past. I had some problems with it, but I'm not sure if they still would be there now. I will tests this and update you in a few days.

> Whatever you do, it seems like the way you specify the variables is not described in the docs. I.e., it is not clear what local_minmax_variables_cons and positivity_variables_cons expects as inputs - this should be documented in the docstring

That's true, thanks.

Reply:
> Why not allow people to say "p" or "rho"?

Is it easily possible to get the index from those variable names? And best, without any hard-coded code for `rho` and `p`.
This would be critical e.g. for multicomponent Euler with more than one density variable.
Optimally, we would allow `density` and `pressure` as well, right? @sloede

Reply:
I implemented a version, where the parameter is a vector of function like `[density]`. It actually worked well, except new allocation in the bounds check routine, which I was not able to fix yet.
Also, this would be very messy for multicomponent Euler for instance, if we would have to define the function before.

Reply:
You can get the variable names and the corresponding indices from `varnames`:
```julia
julia> Trixi.varnames(cons2cons, CompressibleEulerEquations2D(1.4))
("rho", "rho_v1", "rho_v2", "rho_e")

julia> Trixi.varnames(cons2cons, CompressibleEulerMulticomponentEquations2D(; gammas = (1.4, 1.4, 1.4), gas_constants = (2.0, 2.0, 2.0)))
("rho_v1", "rho_v2", "rho_e", "rho1", "rho2", "rho3")
```
Maybe this helps?

Supporting `density` and `pressure` in a general way is difficult, since while I can define a `pressure` function for a system of equations, it will usually not tell me which variable id it is.

My thinking was just that relying too much on indices is a sure way to get into purgatory when users work with multi-species system and have to remember which index needs to change how if they increase the number of species.

Reply:
Thanks. I was looking for some function like `function["rho"] = 1` to do the transfer to Integer efficient. But since this is only done in the constructor once, it probably doesn't have a bis impact.

I now implemented it such that both names and integers are allowed (combined as well):
```
local_minmax_variables_cons_ = Vector{Int}(undef, length(local_minmax_variables_cons))
variables = varnames(cons2cons, equations)
for (i, variable) in enumerate(local_minmax_variables_cons)
    if typeof(variable) == Int
        local_minmax_variables_cons_[i] = variable
        continue
    end
    for (i_, variable_) in enumerate(variables)
        if variable == variable_
            local_minmax_variables_cons_[i] = i_
            break
        end
        error("$variable no valid variable.")
    end
end
```
Is that okay?

Reply:
What's ok or not is really up to the developer, i.e., you :-) However, I would strongly argue against allowing too much flexibility, as this just messes things up. Instead, since you are already supporting indices, why not add a second constructor that allows providing the variables as strings and then convert the strings to integers instead? Mixing integers and strings seems to be overly complicated for something that is likely to not be used by anyone.

But if you feel like having only integer indices is the best way forward, then this is ok as well for me (although I still think string names are much nicer 😬)

Reply:
Is it even possible to use another constructor, if the difference is a keyword argument? I always get error that the function is ambiguous.
No, I think that string names are nicer as well.

Reply:
I realized that I can just set the default value to `Int[]`. That would fix my problems.
Now, it's only an issue if a user passes a general empty vector `[]`.
What do you think?:sweat_smile:

Reply:
Just do what you think is best in terms of usability 🤷

Reply:
Okay, I decided to only allow the variable names as strings. Implemented the change in [3afa4c1](https://github.com/trixi-framework/Trixi.jl/pull/1648/commits/3afa4c1b6a3a331a2acc62433cff4e78f2532b08).